recipes/tree-gen/all/test_package/conanfile.py

```diff
@@ -0,0 +1,31 @@
+import os
+
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, cmake_layout
+
+
+class TreeGenTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualBuildEnv", "VirtualRunEnv"
+    apply_env = False
```

Comment:
```suggestion
```

Not needed in Conan 2, the `env=conanrun` flag in `self.run(bin_path, env="conanrun")` should take care of this